src/backend/distributed/executor/multi_utility.c

```diff
@@ -185,6 +188,53 @@ multi_ProcessUtility(Node *parsetree,
 		return;
 	}
 
+	if (IsA(parsetree, CreateExtensionStmt))
+	{
+		CreateExtensionStmt *createExtensionStmt = (CreateExtensionStmt *) parsetree;
+		if (strcmp(createExtensionStmt->extname, "citus") == 0)
+		{
+			ErrorIfUnstableCreateOrAlterExtensionStmt(parsetree);
+		}
+
+		extensionStatement = true;
+	}
+
+	if (IsA(parsetree, AlterExtensionStmt))
+	{
+		AlterExtensionStmt *alterExtensionStmt = (AlterExtensionStmt *) parsetree;
+		if (strcmp(alterExtensionStmt->extname, "citus") == 0)
+		{
+			ErrorIfUnstableCreateOrAlterExtensionStmt(parsetree);
+		}
+
+		extensionStatement = true;
+	}
+
+	if (IsA(parsetree, DropStmt))
+	{
+		DropStmt *dropStatement = (DropStmt *) parsetree;
+
+		if (dropStatement->removeType == OBJECT_EXTENSION)
+		{
+			extensionStatement = true;
+		}
+	}
+
+	if (extensionStatement)
+	{
+		/*
+		 * In CitusHasBeenLoaded check below, we compare binary Citus version,
+		 * extension version and available version. If they are different, we
+		 * force user to execute ALTER EXTENSION citus UPDATE. Therefore, if
+		 * user executes ALTER EXTENSION or DROP EXTENSION query we should drop
+		 * to the standart utility before CitusHasBeenLoaded check.
+		 */
+		standard_ProcessUtility(parsetree, queryString, context,
```

Comment:
Would it be possible to run this _before_ the `ErrorIfUnstableCreateOrAlterExtensionStmt` calls? In particular, I can run this in my `psql`:

```
jason@[local] ~ ❯❯❯ ALTER EXTENSION citus UPDATE TO '6.3-1';
ERROR:  XX000: requested version is not compatible with loaded Citus binaries.
LOCATION:  ErrorIfUnstableCreateOrAlterExtensionStmt, multi_utility.c:1348
Time: 0.560 ms
```

But there _is_ no Citus 6.3. It should say `ERROR:  22023: extension "citus" has no update path from version "6.2-1" to version "6.3-1"`. By throwing an error first, we mislead the user.